Source/CarthageKit/Project.swift

```diff
@@ -960,6 +960,22 @@ public final class Project {
 			}
 	}
 
+	public func generateSwiftPackageManagerXcodeProjectIfAvailable(forDependencyAt dependencyPath: String) -> SignalProducer<(), NoError> {
+		let files = (try? FileManager.default.contentsOfDirectory(atPath: dependencyPath)) ?? []
+
+		let hasXcodeProjectFile = files.lazy.filter { $0.hasSuffix(".xcodeproj") || $0.hasSuffix(".xcworkspace") }.first != nil
```

Comment:
This will not be sufficient––Carthage allows having Xcode projects or workspaces in any subdirectories so you should need to check the dependency directory recursively.

Reply:
Hmm, something like below might work...
```diff
  public func generateSwiftPackageManagerXcodeProjectIfAvailable(forDependencyAt dependencyPath: String) -> SignalProducer<(), NoError> {
-   let files = (try? FileManager.default.contentsOfDirectory(atPath: dependencyPath)) ?? []
-
-   let hasXcodeProjectFile = files.lazy.first { $0.hasSuffix(".xcodeproj") || $0.hasSuffix(".xcworkspace") } != nil
+   // Search for each sub-dir recursively for Xcode Project file
+   while let fileOrFolder = FileManager.default.enumerator(atPath: dependencyPath)?.nextObject() as? String {
+     let hasXcodeProjectFile = fileOrFolder.hasSuffix(".xcodeproj") || fileOrFolder.hasSuffix(".xcworkspace")
      if hasXcodeProjectFile { return .init(value: ()) }
+   }

+   let files = (try? FileManager.default.contentsOfDirectory(atPath: dependencyPath)) ?? []
    let hasPackageFile = files.lazy.first { $0 == "Package.swift" } != nil
    guard hasPackageFile else { return .init(value: ()) }
```